registration

evacalla opened the pull request:
changed logo.png
added new route routes.dart.
added folder registration
added file registration_page.dart, I used smart_select easily convert from select.
added file resgistration_from.dart
added file index.dart
added new dependency smart_select, reference => https://pub.dev/packages/smart_select

evacalla committed:
registration 

changed logo.png
added new route routes.dart.
added folder registration
added file registration_page.dart, I used smart_select easily convert from select.
added file resgistration_from.dart
added file index.dart
added new dependece smart_select, reference => https://pub.dev/packages/smart_select

matiasm15 commented on lib/screens/login/widgets/login_form.dart:
```diff
@@ -70,6 +70,10 @@ class _LoginFormState extends State<LoginForm> {
   }
 
+  void _formRegistration() {
+    Navigator.pushReplacementNamed(context, '/registration');
```
Usaria `Navigator.push(context, '/registration');` para que al hacer click en el boton de atras vuelvas al login en vez de cerrar la app.

evacalla replied:
Ok

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+  List<S2Choice<int>> restriction = [
+    S2Choice<int>(value: 1, title: 'Vegetariano'),
+    S2Choice<int>(value: 2, title: 'Vegano'),
+    S2Choice<int>(value: 3, title: 'Celiaco'),
```
Habiamos dicho de usar solo vegetariano o vegano. No creo que en el back hagan un endpoint para eso.

evacalla replied:
En algún momento si se piensa agregar más restricciones estaría bueno que nos brinde back un endpoint, otra cosa el 'Celiaco' no estaba para pintar

matiasm15 replied:
Si, pero para el MVP no tiene sentido si son siempre los mismos 2 valores.

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+          width: double.infinity,
+          height: 40,
+          child: LoadingButton(
+            text: 'Registar',
```
`Registrar`

evacalla replied:
Es que vos no sabes esperanto

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+              child: SmartSelect.single(
+                  value: _selected,
+                  choiceItems: restriction,
+                  modalTitle: 'Restrición Alimenticia',
```
`Restricción`

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+      Timer(const Duration(seconds: 3), () {
+
+        if(_selected == null || _selected.isEmpty) {
+          uiStore.setErrorMessage("Debe selecionar una restricción");
```
`seleccionar`

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+          width: double.infinity,
+          height: 60,
+          child: TextButton(
+            child: const Text('¿Tenes una cuenta?'),
```
Tenés

matiasm15 commented on lib/screens/login/widgets/login_form.dart:
```diff
@@ -131,7 +135,15 @@ class _LoginFormState extends State<LoginForm> {
+          width: double.infinity,
+          height: 40,
+          child: TextButton(
+              child: const Text('Registrate'),
```
Desentona bastante el color azul para este link, quizas un gris como en el mock?

matiasm15 commented on lib/screens/registration/registration_page.dart:
```diff
@@ -0,0 +1,22 @@
+  Widget build(BuildContext context) {
+    return BasicScaffold(
+      backgroundColor: Colors.yellow,
+      body: Column(children: <Widget>[
```
En mi compu el logo queda muy grande lo que hacer que tenga que scrollear para ver toda la pantalla, deberia tener el mismo tamaño que en el logo de la pantalla de login.

evacalla replied:
Comprate un celular nuevo asi queda bien 👍

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+
+class _RegistrationFormState extends State<RegistrationForm> {
+
+  String _email, _password, _passwordRepeat, _selected;
```
Re generico el wording `_selected`, que palabra habran usado en el back? `_restricction`?

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+  String _comparePassword(String value) {
+
+    if(value.isEmpty) {
+      return 'El contraseña no puede ser vacío';
```
La contraseña

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+    }
+
+    if(!identical(_password, value)) {
+      return 'El contraseña debe coincidir';
```
La contraseña

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+  Widget build(BuildContext context) {
+    return Form(
+      key: _formKey,
+      child: Column(mainAxisAlignment: MainAxisAlignment.center, children: <Widget>[
```
Falta el nombre de la persona.

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,228 @@
+              padding: const EdgeInsets.only(bottom: 15),
+              child: Focus(
+                child: TextFormField(
+                  enableSuggestions: false,
```
Yo los deje habilitado `enableSuggestions` y `autocorrect` en el login para el email, le permite al usuario que Google le sugiera su email si lo usa en varias apps y no veo un riesgo de seguridad como si en la password.

matiasm15 committed:
Delete html import

evacalla committed:
registration | changed by pr

Changed text and color
Added field userName in form registration

evacalla committed:
Merge remote-tracking branch 'origin/registration' into registration

evacalla committed:
registration - add lib to doc readme

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+                padding: const EdgeInsets.only(bottom: 15),
+                child: Focus(
+                  child: TextFormField(
+                    enableSuggestions: true,
```
Por seguridad no te tiene que sugerir ni corregir la contraseña.
``` 
                   enableSuggestions: false,
                    autocorrect: false,
```

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+                  child: TextFormField(
+                    enableSuggestions: true,
+                    autocorrect: true,
+                    obscureText: true,
```
Por que `obscureText: true`? jajaj

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+                    obscureText: true,
+                    onChanged: _userNameChanged,
+                    validator: _userNameValidator,
+                    decoration: const InputDecoration(hintText: 'Usuario', icon: Icon(Icons.person)),
```
Lo cambiaria por `Nombre`, `Usuario` da a entender que es un username.

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+      return 'El usuario no puede ser vacío';
+    }
+
+    if(value.length < 8 && value.length > 25) {
```
Para que un limite minimo tan alto?

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+    }
+
+    if(!identical(_password, value)) {
+      return 'La contraseña debe coincidir';
```
No me deja registrarme porque me tira siempre este error GG.

evacalla committed:
registration | add size imagen

It adds size image
It removes atribute obscureText y enableSuggestion, autocorrector changes value false
It changes name from usuario to nombre

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+    }
+
+    if(value.length > 40) {
+      return 'El nombre debe tener máximo 25 caracteres';
```
'El nombre debe tener máximo 40 caracteres'

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+  String _passwordValidator(String value) {
+
+    if(value.isEmpty) {
+      return 'La contraseña no puede ser vacia';
```
La contraseña no puede ser vacía

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+  String _comparePassword(String value) {
+
+    if(value.isEmpty) {
+      return 'La contraseña no puede ser vacío';
```
La contraseña no puede ser vacía

matiasm15 commented on lib/screens/registration/widgets/registration_form.dart:
```diff
@@ -0,0 +1,261 @@
+    }
+
+    if(value.length < 8) {
+      return 'La contraseña debe tener minímo 8 caracteres';
```
La contraseña debe tener mínimo 8 caracteres

matiasm15 committed:
Revert "Login users"

This reverts commit a85c9e0d90f94b2f7dcf441256990e3d36bb5cbe.

matiasm15 committed:
Login users

matiasm15 committed:
Remove rule

evacalla committed:
registration | changed message error

evacalla committed:
Merge remote-tracking branch 'origin/registration' into registration

evacalla committed:
registration | correction dart analysis
